feat: Add getDatabaseInfo method to query database Information including table names.

Darshan1904 opened the pull request.

Darshan1904 committed:
feat: Add getDatabases method to query database names

Darshan1904 committed:
Merge branch 'master' into darshan

Darshan1904 committed:
feat: Change getDatabaseNames function to getDatabaseInfo inorder to get information about tables present in databases

ragpatel8742 commented on src/AlfangoDB/modules/read.mo:
```diff
@@ -179,4 +182,91 @@ module {
+        ] = [];
+
+        // crete an Iterator
+        let databaseArgsIter = Iter.range(0, alfangoDB.databases.size() - 1);
```
1. Do not use nested switch cases, instead use `?do{}` (option block)
2. Instead of Array.append() use Buffer https://internetcomputer.org/docs/current/motoko/main/base/Buffer

ragpatel8742 replied:
Refer src/AlfangoDB/modules/read.mo to understand usage of _option block_ and _Buffer_.

Darshan1904 committed:
Refactor getDatabaseInfo function to improve readability and efficiency

- Replaced nested switch cases with ?do option blocks for cleaner handling of Option values.
- Switched from using Array.append() to Buffer.

Darshan1904 committed:
Remove unnecessary imports

Darshan1904 committed:
Refactor getDatabaseInfo for improved efficiency and readability

- Replace nested iterations with Map.entries() for direct access to custom hash map
- Remove unnecessary nested option blocks
- Flatten structure to two main loops for databases and tables.

ragpatel8742 commented on src/AlfangoDB/modules/read.mo:
```diff
@@ -179,4 +180,29 @@ module {
         Prelude.unreachable();
     };
 
+    public func getDatabaseInfo(alfangoDB: Database.AlfangoDB) : OutputTypes.GetDatabasesOutputType {
```
Lastly, let us change this function name to GetDatabases in order to make it with input and output type names.

Darshan1904 committed:
Rename getDatabaseInfo to getDatabases inorder to match input and output type names.
